gadgets/trace_malloc/program.bpf.c

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright (c) 2024 The Inspektor Gadget authors */
+
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <gadget/buffer.h>
+#include <gadget/macros.h>
+
+struct event {
+	__u32 pid;
+	__u64 addr;
+	bool is_malloc;
+};
+
+GADGET_TRACER_MAP(events, 1024 * 256);
+
+GADGET_TRACER(open, events, event);
+
+SEC("uretprobe//usr/lib/libc.so.6:malloc")
+int trace_uprobe_malloc(struct pt_regs *ctx)
+{
+	struct event *event;
+
+	event = gadget_reserve_buf(&events, sizeof(*event));
+	if (!event)
+		return 0;
+
+	event->pid = bpf_get_current_pid_tgid() >> 32;
+	event->addr = PT_REGS_RC(ctx);
+	event->is_malloc = 1;
+
+	gadget_submit_buf(ctx, &events, event, sizeof(*event));
+
+	return 0;
+}
+
+SEC("uprobe//usr/lib/libc.so.6:free")
+int trace_uprobe_free(struct pt_regs *ctx)
+{
+	struct event *event;
+
+	event = gadget_reserve_buf(&events, sizeof(*event));
+	if (!event)
+		return 0;
+
+	event->pid = bpf_get_current_pid_tgid() >> 32;
+	event->addr = PT_REGS_PARM1(ctx);
+	event->is_malloc = 0;
+
+	gadget_submit_buf(ctx, &events, event, sizeof(*event));
+
+	return 0;
+}
```

Comment:
nit: perhaps you can create a helper function that does all the work and receives a `is_malloc` argument? Then these two programs will use the same function.